### ISSUE TITLE
CUDA Capability sm_90 for H100 GPUs

### ISSUE THREAD

**Issue by Schorob** [pytorch/pytorch#90761]
### 🚀 The feature, motivation and pitch

Hello, 

i would like to ask when PyTorch will support the sm_90 CUDA capability. My request is motivated by the necessity of this compute capability to use H100 GPUs. 

When i try to send a tensor to a H100 GPU, the following warning is given: 

/home/schoerob/cuda_stuff/testnev/lib/python3.8/site-packages/torch/cuda/__init__.py:155: UserWarning:
NVIDIA H100 PCIe with CUDA capability sm_90 is not compatible with the current PyTorch installation.
The current PyTorch install supports CUDA capabilities sm_37 sm_50 sm_60 sm_70 sm_75 sm_80 sm_86.
If you want to use the NVIDIA H100 PCIe GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/

  warnings.warn(incompatible_device_warn.format(device_name, capability, " ".join(arch_list), device_name))

Thank you in advance. 

cc @ezyang @seemethere @malfet @ngimel

**Comment by soumith** [comment]
cc: @ptrblck

**Comment by ptrblck** [comment]
My PR enabling nightly binary builds with CUDA 11.8 is open, but blocked on the `1.13.1` release. Once the release is done, the PR will be reviewed and we can start with nightly builds. A follow up PR will the enable sm_90 for these builds.

**Comment by DP1701** [comment]
Is there any news? I have the same problem with PyTorch 2.0 using CUDA V12.0.140 or V11.8.89.

**Comment by ptrblck** [comment]
Hopper support was added so I'll close this issue. Please reopen it or create a new issue tagging me in case you see any issues.

**Comment by ShrutiSarikaChakraborty** [comment]
Hi,

I am getting this warning message - "NVIDIA H100 PCIe with CUDA capability sm_90 is not compatible with the current PyTorch installation." and this error "RuntimeError: CUDA error: no kernel image is available for execution on the device
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
".

Everything was working till last week. I don't understand what is causing this issue. Can you please help.Thanks!

**Comment by thegallier** [comment]
same issue

**Comment by alexpancheva** [comment]
Hi @ptrblck we seem to be having an issue here. See the questions above. Thanks

**Comment by alexpancheva** [comment]
The only setup this works on for me is pytorch nightly. Conda install

**Comment by ptrblck** [comment]
> conda installation seems to work. The suggested pip installation does not. Ubuntu.

All pip wheels using CUDA 11.8+ work on H100:
```python
import torch

print(torch.__version__)
print(torch.cuda.get_device_properties(0))

print(torch.randn(1, device="cuda"))
```
```
pip install torch --index-url https://download.pytorch.org/whl/cu118
2.0.1+cu118
_CudaDeviceProperties(name='NVIDIA H100 80GB HBM3', major=9, minor=0, total_memory=81082MB, multi_processor_count=132)
tensor([-0.9087], device='cuda:0')

pip install --pre torch --index-url https://download.pytorch.org/whl/nightly/cu118
2.1.0.dev20230810+cu118
_CudaDeviceProperties(name='NVIDIA H100 80GB HBM3', major=9, minor=0, total_memory=81082MB, multi_processor_count=132)
tensor([-0.6681], device='cuda:0')

pip install --pre torch --index-url https://download.pytorch.org/whl/nightly/cu121
2.1.0.dev20230810+cu121
_CudaDeviceProperties(name='NVIDIA H100 80GB HBM3', major=9, minor=0, total_memory=81082MB, multi_processor_count=132)
tensor([0.1740], device='cuda:0')
```
Posting "same issue" without any more information is not actionable.

**Comment by shashank140195** [comment]
> Hi,
> 
> I am getting this warning message - "NVIDIA H100 PCIe with CUDA capability sm_90 is not compatible with the current PyTorch installation." and this error "RuntimeError: CUDA error: no kernel image is available for execution on the device CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect. For debugging consider passing CUDA_LAUNCH_BLOCKING=1. Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions. ".
> 
> Everything was working till last week. I don't understand what is causing this issue. Can you please help.Thanks!

@ShrutiSarikaChakraborty I am getting the same issue, how did you resolve it?

**Comment by ptrblck** [comment]
@shashank140195 post a minimal and executable code snippet as well as the output of the `collect_env` script as all current binaries support the H100.
